esp-wifi/src/lib.rs

```diff
@@ -396,12 +370,11 @@ impl EspWifiTimerSource for TimeBase {
 /// .unwrap();
 /// # }
 /// ```
-pub fn initialize(
-    init_for: EspWifiInitFor,
-    timer: impl EspWifiTimerSource,
+pub fn initialize<'d, T: EspWifiTimerSource>(
+    mut timer: impl Peripheral<P = T> + 'd,
     rng: hal::rng::Rng,
```

Comment:
Should we also take rng as peripheral ref?

Reply:
I'd say no since the rng driver can be cloned.

Reply:
That might actually be a slight problem since we're not even paying any attention to the peripheral's suggested read frequency (ESP32 suggests not reading faster than 500ksps). If we start to do that, we might want to introduce some sort of shared access around the peripheral which may need a lock (on multiple cores) or some other synchronisation, which is not necessary if we know there is a single user.

Reply:
The rate limit only matters if you want true random numbers, which I guess the WiFi driver wants? (Idk much about this driver) Maybe having a mix is acceptable? This PR is probably not the place to start tweaking this.

+1 on the locking stuff, might even be able to get away with atomics since it's just throttling at the end of the day.

Reply:
I've changed `init` to allow `Rng` or `Trng`, still not using peripheral ref as Trng isn't safe to clone (we need some ref counting see the discussion in https://github.com/esp-rs/esp-hal/issues/2372)